### PR TITLE
Custom Resource of iis_site

### LINKED ISSUE #318: Uninitialized constant Chef::Resource::IisSite
### Cookbook version
windows => 3.0.3
iis => 5.1.0

### Chef-client version
Chef Client 12.19.36

### Platform Details
Windows Server 2012 R2

### Scenario:
i am trying to install a new web site using the iis_site resource

### Steps to Reproduce:
1. Create a cookbook with a resource for iis_site
2. Execute chef-client

### Expected Result:
Web site is created

### Actual Result:
I am getting the error:

NameError
---------
uninitialized constant Chef::Resource::IisSite

### PR DESCRIPTION
**Pull request by EasyAsABC123**
Signed-off-by: Justin Schuhmann <jmschu02@gmail.com>

### Description

Converts the provider to a custom resource

### Issues Resolved

#318

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [X] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [X] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: resources/site.rb [hunk 1]
```diff
@@ -18,20 +17,211 @@
...
-attribute :log_truncsize, kind_of: Integer, default: 1_048_576
-
-attr_accessor :exists, :running
+require 'rexml/document'
+
+include REXML
+include Opscode::IIS::Helper
+include Opscode::IIS::Processors
+
+property    :site_name, String, name_attribute: true
```
Comment (iennae) [inline resources/site.rb]: name_property replaces name_attribute

File: resources/site.rb [hunk 2]
```diff
@@ -18,20 +17,211 @@
...
+      # support for additional options -logDir, -limits, -ftpServer, etc...
+      cmd << " #{new_resource.options}" if new_resource.options
+      shell_out!(cmd, returns: [0, 42])
+
+      configure
+
+      if new_resource.application_pool
+        shell_out!("#{appcmd(node)} set site /site.name:\"#{new_resource.site_name}\" /[path='/'].applicationPool:\"#{new_resource.application_pool}\"", returns: [0, 42])
+      end
+      new_resource.updated_by_last_action(true)
```
Comment (iennae) [inline resources/site.rb]: we should never be using the updated_by_last_action with modern chef code, instead should be doing converge_by blocks.

File: resources/site.rb [hunk 3]
```diff
@@ -18,20 +17,213 @@
...
+  else
+    Chef::Log.debug("#{new_resource} site does not exist - nothing to do")
+  end
+end
+
+action :start do
+  if !current_resource.running
+    converge_by "Started the Site - \"#{new_resource}\"" do
+      shell_out!("#{appcmd(node)} start site /site.name:\"#{new_resource.site_name}\"", returns: [0, 42])
+      new_resource.updated_by_last_action(true)
```
Comment (iennae) [inline resources/site.rb]: updated_by_ needs to be removed here.

**Comment by EasyAsABC123** [comment]
@iennae nice, i must of missed that one...weird

**Pull request merged**